### PR TITLE
Complete To Do's of new user guide

### PR DESCRIPTION
**Pull request by LLdaniel**
After merging the new user guide this PR is for completing the open To Do`s of the user guide. 
This will help to finish #106. 

The goal is to document the following:
- [x] complete motivation letter section
- [x] explanation of adding pictures
- [x]  note about handling long urls and names (the to do sounds like there is a procedure which I am not aware of. Maybe my current part should be reworked then)
- [x]  add more adjustable length
- [x]  custom styles by combining headers, footers, bodies etc.
- [x] manually compare old userguide for not missing out important information
- [x] transfer missing comments in `template.tex` to the new userguide (imho: already done with the inital rework, nothing to do here)

EDIT: Tasklist progress

### CODE REVIEW

File: manual/moderncv_userguide.tex
Comment (tianyizheng02) [inline manual/moderncv_userguide.tex:781]:
```suggestion
In case you do not want \texttt{symbols} (default) for the icons, there is also the option to replace icons with \texttt{letters} instead. This is set by the macro \code{\\moderncvicons}, which must be called after setting the style in the preamble.
```

File: manual/moderncv_userguide.tex
Comment (tianyizheng02) [inline manual/moderncv_userguide.tex:785]:
```suggestion
This mechanism is also used to determine the icon set. In principle, the \code{\\moderncvicons} can be used to determine the icon set with the possible values \texttt{marvosym}, \texttt{awesome} or \texttt{academic}, but it is recommended to let {\moderncv} decide the correct icon set due to differing \Latex compiler support.
```

File: manual/moderncv_userguide.tex
Comment (tianyizheng02) [inline manual/moderncv_userguide.tex:816]:
```suggestion
The spacing between the hints column and the CV entries can be adjusted by setting\\\code{\\separatorcolumnwidth}. The default value is \code{0.025\\textwidth}.
```
The line exceeds the margin in the PDF, so I added a manual line break.

File: manual/moderncv_userguide.tex
Comment (tianyizheng02) [inline manual/moderncv_userguide.tex:821]:
```suggestion
The width of the footer can be adjusted by setting \code{\\footwidth}. Due to internal calculations \code{\\renewcommand} is needed. The default value is \code{0.8\\textwidth}.
```

File: manual/moderncv_userguide.tex
Comment (tianyizheng02) [inline manual/moderncv_userguide.tex:827]:
```suggestion
The width of the quote can be adjusted by setting \code{\\quotewidth}. Due to internal calculations \code{\\renewcommand} is needed. The default value is \code{0.65\\textwidth}.
```

File: manual/moderncv_userguide.tex
Comment (tianyizheng02) [inline manual/moderncv_userguide.tex:875]:
```suggestion
To create links to a website or a email address, use the following commands:

A general weblink with optional text.  
\begin{lstlisting}
  \link[<text>]{<link>}
  %% example
  \link[name of the link]{ftp://ftp.somesite.org}
\end{lstlisting}
```

File: manual/moderncv_userguide.tex
Comment (tianyizheng02) [inline manual/moderncv_userguide.tex:882]:
```suggestion
An HTTP link. The HTTP prefix is generated automatically and is not needed in the link argument.
\begin{lstlisting}
  \link[<text>]{<link>}
  %% example
  \link[goto HTTP site]{www.somehttpsite.org}
\end{lstlisting}
```

File: manual/moderncv_userguide.tex
Comment (tianyizheng02) [inline manual/moderncv_userguide.tex:889]:
```suggestion
An HTTPS link. The HTTPS prefix is generated automatically and is not needed in the link argument.
\begin{lstlisting}
  \link[<text>]{<link>}
  %% example
  \httpslink[goto HTTPS site]{www.somesecuresite.org}
\end{lstlisting}
```

File: manual/moderncv_userguide.tex
Comment (tianyizheng02) [inline manual/moderncv_userguide.tex:891]:
```suggestion
An email link. The mailto prefix is generated automatically and is not needed in the link argument.
```
"email" rather than "e-mail" is consistently used throughout the rest of the doc

File: manual/moderncv_userguide.tex
Comment (tianyizheng02) [inline manual/moderncv_userguide.tex:964]:
```suggestion
Currently {\moderncv} ships with five different styles: \texttt{casual} (default), \texttt{classic}, \texttt{banking}, \texttt{oldstyle} and \texttt{fancy}. Internally, these styles are numbered 1, 2, 3, 4 and 5, respectively. All styles except \texttt{oldstyle} and \texttt{fancy} possess additional option parameters which were described in section \ref{section:customization:stylesAndOptions}.
```

File: manual/moderncv_userguide.tex
Comment (tianyizheng02) [inline manual/moderncv_userguide.tex:966]:
```suggestion
Different styles' head, body and foot can be combined to create custom styles. This is done by setting \code{\\moderncvhead}, \code{\\moderncvbody} and \code{\\moderncvfoot} in the preamble. Here is an example of a custom style with a \textt{banking} head, an \textt{oldstyle} body and a \textt{casual} foot:
```

File: manual/moderncv_userguide.tex
Comment (tianyizheng02) [inline manual/moderncv_userguide.tex:972]:
```suggestion
Furthermore, options for each part of {\moderncv} can be set as well. To get a CV with a \texttt{classic} head and a \texttt{banking} body with full rules, define the following in the preamble:
```

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]
I have some suggestions on phrasing, but overall it looks good! I think all of the TODOs have been more or less addressed, though I'm not particularly familiar with the original author's intent behind the TODOs.

File: manual/moderncv_userguide.tex
Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: Corrected with [303fd96](https://github.com/moderncv/moderncv/pull/184/commits/303fd9693cde6846ac9b79cf77368aba58252ed0)

Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: Ah, didn´t know about that, but I agree:
Maybe we should open a separate issue?

File: manual/moderncv_userguide.tex
Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: Corrected with [303fd96](https://github.com/moderncv/moderncv/pull/184/commits/303fd9693cde6846ac9b79cf77368aba58252ed0)

File: manual/moderncv_userguide.tex
Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: Corrected with [303fd96](https://github.com/moderncv/moderncv/pull/184/commits/303fd9693cde6846ac9b79cf77368aba58252ed0)

File: manual/moderncv_userguide.tex
Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: Corrected with [303fd96](https://github.com/moderncv/moderncv/pull/184/commits/303fd9693cde6846ac9b79cf77368aba58252ed0)

File: manual/moderncv_userguide.tex
Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: hehe, you are right. Better do not translate things from your native language directly... ^^
Thanks for correcting this!
Corrected with [303fd96](https://github.com/moderncv/moderncv/pull/184/commits/303fd9693cde6846ac9b79cf77368aba58252ed0)

File: manual/moderncv_userguide.tex
Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: Corrected with [303fd96](https://github.com/moderncv/moderncv/pull/184/commits/303fd9693cde6846ac9b79cf77368aba58252ed0)

File: manual/moderncv_userguide.tex
Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: Corrected with [303fd96](https://github.com/moderncv/moderncv/pull/184/commits/303fd9693cde6846ac9b79cf77368aba58252ed0)

File: manual/moderncv_userguide.tex
Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: Corrected with [303fd96](https://github.com/moderncv/moderncv/pull/184/commits/303fd9693cde6846ac9b79cf77368aba58252ed0)

File: manual/moderncv_userguide.tex
Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: Corrected with [303fd96](https://github.com/moderncv/moderncv/pull/184/commits/303fd9693cde6846ac9b79cf77368aba58252ed0)

File: manual/moderncv_userguide.tex
Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: Corrected with [303fd96](https://github.com/moderncv/moderncv/pull/184/commits/303fd9693cde6846ac9b79cf77368aba58252ed0)

File: manual/moderncv_userguide.tex
Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: Corrected with [303fd96](https://github.com/moderncv/moderncv/pull/184/commits/303fd9693cde6846ac9b79cf77368aba58252ed0)

File: manual/moderncv_userguide.tex
Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: Corrected with [303fd96](https://github.com/moderncv/moderncv/pull/184/commits/303fd9693cde6846ac9b79cf77368aba58252ed0)

File: manual/moderncv_userguide.tex
Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: Corrected with [303fd96](https://github.com/moderncv/moderncv/pull/184/commits/303fd9693cde6846ac9b79cf77368aba58252ed0)

File: manual/moderncv_userguide.tex
Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: Corrected with [303fd96](https://github.com/moderncv/moderncv/pull/184/commits/303fd9693cde6846ac9b79cf77368aba58252ed0)

File: manual/moderncv_userguide.tex
Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: Corrected with [303fd96](https://github.com/moderncv/moderncv/pull/184/commits/303fd9693cde6846ac9b79cf77368aba58252ed0)

File: manual/moderncv_userguide.tex
Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: Corrected with [303fd96](https://github.com/moderncv/moderncv/pull/184/commits/303fd9693cde6846ac9b79cf77368aba58252ed0)

File: manual/moderncv_userguide.tex
Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: Corrected with [303fd96](https://github.com/moderncv/moderncv/pull/184/commits/303fd9693cde6846ac9b79cf77368aba58252ed0)

File: manual/moderncv_userguide.tex
Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: Corrected with [303fd96](https://github.com/moderncv/moderncv/pull/184/commits/303fd9693cde6846ac9b79cf77368aba58252ed0)

**Comment by LLdaniel** [comment]
Thank you very much @tianyizheng02 for your review. I appreciate your second pair of eyes and tried to address all these corrections in [303fd96](https://github.com/moderncv/moderncv/pull/184/commits/303fd9693cde6846ac9b79cf77368aba58252ed0).

**Comment by jalopezg-git** [comment]
Hey @LLdaniel, apologies for noticing this now.  I'll try to review throughout the week :slightly_smiling_face:!

File: manual/moderncv_userguide.tex
Comment (jalopezg-git) [inline manual/moderncv_userguide.tex:969]:
```suggestion
Currently {\moderncv} ships with six different styles: \texttt{casual} (default), \texttt{classic}, \texttt{banking}, \texttt{oldstyle}, \texttt{fancy} and \texttt{contemporary}. Internally, these styles are numbered 1, 2, 3, 4, 5 and 6, respectively. All styles except \texttt{oldstyle} and \texttt{fancy} possess additional option parameters which were described in section \ref{section:customization:stylesAndOptions}.
```

File: manual/moderncv_userguide.tex
Comment (jalopezg-git) [inline manual/moderncv_userguide.tex:971]:
```suggestion
Different styles' head, body and foot can be combined to create custom styles. This is done by setting \code{\\moderncvhead}, \code{\\moderncvbody} and \code{\\moderncvfoot} in the preamble. Here is an example of a custom style with a \texttt{banking} head, an \texttt{oldstyle} body and a \texttt{casual} foot:
```

**Review by jalopezg-git — CHANGES_REQUESTED** [review]
Just a couple of tiny, small changes!  Once fixed, it's ready to go in!  Many thanks, @LLdaniel!

File: manual/moderncv_userguide.tex
Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: Ah, yes, in the meantime we already have six cool styles... Forgot that passage here, thanks for mentioning it!
Is now resolved with [e0e9259](https://github.com/moderncv/moderncv/pull/184/commits/e0e9259201620d26f233190f0d366627be653b81)

File: manual/moderncv_userguide.tex
Comment (LLdaniel) [reply manual/moderncv_userguide.tex]: Hmm... Somebody stole a `D` here. Thanks! Is now resolved with [e0e9259](https://github.com/moderncv/moderncv/pull/184/commits/e0e9259201620d26f233190f0d366627be653b81).

**Comment by LLdaniel** [comment]
Hi @jalopezg-git ,
thank you for your code review! This helps a lot to finish a part of the set milestone. :tada: 

I have to admit that I had some issues with rebasing and did a merge now. Hopefully this is also ok. It was a bit difficult to handle the .pdf here. Therefore I simply took "ours" and recompiled it after the merge. 
Now it should be the latest version.

**Comment by tianyizheng02** [comment]
Given that this PR more or less "completes" the user guide, does this mean we can finally create a new tag and upload it to CTAN?

**Pull request merged**
